### PR TITLE
Support DECIMAL and remove DOUBLE in TPCH Connector

### LINKED ISSUE #3557: Decimal support in TPCH
Hi Guys,

As you know we have been working on decimal type for some time. We have implemented decimal using both `long` primitive (for short decimals) and `Slice` (for long decimals). We would like to benchmark decimal type using presto benchmarking mechanism. In order to do so it would be best to use decimal types in TPCH connector. We propose to introduce additional schemas in TPCH connector. We would have three schemas families: one using short decimals (precision <= 17), one using long decimals (precision > 17) and one using double types for numeric TPCH columns. Such approach would enable us to test decimal performance using in memory TPCH connector and in hive connector. It would also enable us to check compatibility with TPCH test suite, since all required types would be supported.

CC: @martint @losipiuk @cberner

### PR DESCRIPTION
**Pull request by atris**
Fixes #3557 

Airlift PR: https://github.com/airlift/tpch/pull/23

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,9 +5,17 @@
     <parent>
```
Comment (atris) [inline pom.xml]: Changes made to this file will be reverted prior to merging. Please ignore these changes

File: pom.xml [hunk 2]
```diff
@@ -5,9 +5,17 @@
...
         <groupId>io.airlift</groupId>
         <artifactId>airbase</artifactId>
-        <version>80</version>
```
Comment (kokosing) [inline pom.xml]: it looks like you depends on other PR which is in progress, it should be a separate commit (cherry-picked)

File: pom.xml [hunk 3]
```diff
@@ -5,9 +5,17 @@
...
+        <version>82</version>
     </parent>
 
+    <repositories>
+        <repository>
+            <id>atri-local</id>
+            <name>Local AirLift Repo</name>
+            <url>/Users/atrisharma/tpch/target</url>
+        </repository>
+    </repositories>
```
Comment (kokosing) [inline pom.xml]: ?

File: pom.xml [hunk 4]
```diff
@@ -598,12 +611,6 @@
                 <version>${dep.drift.version}</version>
             </dependency>
 
-            <dependency>
-                <groupId>io.airlift.tpch</groupId>
-                <artifactId>tpch</artifactId>
-                <version>0.9</version>
-            </dependency>
-
```
Comment (kokosing) [inline pom.xml]: do not move

File: presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/TableStatisticsRecorder.java [hunk 5]
```diff
@@ -72,8 +72,8 @@
                 return column.getInteger(row);
             case DATE:
                 return column.getDate(row);
-            case DOUBLE:
-                return column.getDouble(row);
+            case DECIMAL:
+                return column.getIdentifier(row);
```
Comment (kokosing) [inline presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/TableStatisticsRecorder.java]: did you mean `long`?

File: presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java [hunk 6]
```diff
@@ -487,8 +491,8 @@ public static Type getPrestoType(TpchColumn<?> column)
                 return INTEGER;
             case DATE:
                 return DATE;
-            case DOUBLE:
-                return DOUBLE;
+            case DECIMAL:
+                return DECIMAL;
```
Comment (kokosing) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: you should return DECIMAL(12,2), use com.facebook.presto.spi.type.DecimalType#createDecimalType(int, int)

File: pom.xml (hunk 2 above)
Comment (atris) [reply pom.xml]: Please see:
https://github.com/prestodb/presto/pull/11166#discussion_r206802469

It is hard for me to locally separate out this commit, and my local testing needs these changes. Hence I raised this as part of the PR, will revert before merging. Please ignore any changes to this file

File: presto-spi/src/main/java/com/facebook/presto/spi/type/DecimalType.java [hunk 7]
```diff
@@ -29,6 +29,7 @@
 {
     public static final int DEFAULT_SCALE = 0;
     public static final int DEFAULT_PRECISION = MAX_PRECISION;
+    public static final DecimalType DECIMAL = createDecimalType(12, 2);
```
Comment (kokosing) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/DecimalType.java]: please undo

File: pom.xml (hunk 2 above)
Comment (atris) [reply pom.xml]: Please see:
https://github.com/prestodb/presto/pull/11166#discussion_r206802469

It is hard for me to locally separate out this commit, and my local testing needs these changes. Hence I raised this as part of the PR, will revert before merging. Please ignore any changes to this file

File: presto-spi/src/main/java/com/facebook/presto/spi/type/DecimalType.java (hunk 7 above)
Comment (atris) [reply presto-spi/src/main/java/com/facebook/presto/spi/type/DecimalType.java]: Types usually have a static created type (e.g. DOUBLE etc), hence I created one for DECIMAL. I will use ``createDecimal`` everywhere instead

File: pom.xml (hunk 3 above)
Comment (atris) [reply pom.xml]: https://github.com/prestodb/presto/pull/11166#discussion_r206802469

File: pom.xml (hunk 4 above)
Comment (atris) [reply pom.xml]: https://github.com/prestodb/presto/pull/11166#discussion_r206802469

File: presto-spi/src/main/java/com/facebook/presto/spi/type/DecimalType.java (hunk 7 above)
Comment (atris) [reply presto-spi/src/main/java/com/facebook/presto/spi/type/DecimalType.java]: I was following the usual pattern of types having a static instance which can be returned. I will use ``createDecimal``

File: presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java (hunk 6 above)
Comment (atris) [reply presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: Ack

File: presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/TableStatisticsRecorder.java (hunk 5 above)
Comment (atris) [reply presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/TableStatisticsRecorder.java]: ``getIdentifier`` returns the long value of the column, should it not work for this case?

**Comment by sopel39** [comment]
I think some classes in `presto-benchmark/src/main/java/com/facebook/presto/benchmark` need to be modified/removed as well. 

You might want to try running: `com.facebook.presto.benchmark.TestBenchmarks`
